baker/bakeSiteOnStagingServer.ts

```diff
@@ -1,9 +1,33 @@
 // todo: remove this file
 
 import { bake } from "./DeployUtils.js"
-bake().then((_) => {
-    // TODO: without this the script hangs here since using the workerpool library in baking
-    // I don't understand why this happens. Probably using top level await would also resolve
-    // this but I couldn't get Typescript to play along with that
-    process.exit(0)
-})
+import yargs from "yargs"
+import { hideBin } from "yargs/helpers"
+import { bakeSteps, validateBakeSteps } from "./SiteBaker.js"
+
+yargs(hideBin(process.argv))
+    .command<{
+        steps?: string[]
+    }>(
+        "$0",
+        "Deploy the site to a remote environment",
+        (yargs) => {
+            yargs.option("steps", {
+                type: "array",
+                choices: bakeSteps,
+                description: "Steps to perform during the baking process",
+            })
+        },
+        async ({ steps }) => {
+            const bakeSteps = validateBakeSteps(steps)
```

Comment:
Nit - yargs does the validation and raises something like 

```
validateBakeStepsInvalid values:
  Argument: steps, Given: "test", Choices: "assets", "blogIndex", "embeds", "googleScholar", "redirects", "rss", "variables", "wordpressPosts", "specialPages", "countries", "countryProfiles", "charts", "gdocPosts", "gdriveImages"
```

so you don't need `validateBakeSteps`

Reply:
yargs gives us `string[]` when I wanted `BakeStep[]` and afaik there's no way to get it to play nicer with TS, hence the extra step.

But you're right, it's overkill. I'll just cast `string[] as BakeSte[]`